lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -3564,6 +3564,74 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
             binder.op, resultType, permutedStft);
         return success();
       });
+
+  // split to sequence
+  // Arguments:
+  // - input: the tensor to split
+  // -Split(optional): Length of each output
+  // Attributes:
+  // - axis: the axis along which to split the input
+  // - keepdims: to keep the split dimension or not. Ignored when 'split' is
+  // specified Outputs:
+  // - outputs: sequence of tensor
+  //
+
+  patterns.onOp(
+      "SplitToSequence", 11,
+      [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Value self;
+        Value split;
+        int64_t axis;
+        int64_t keepdims;
+        Torch::ListType resultType;
+
+        if (binder.tensorOperandAtIndex(self, 0))
+          return rewriter.notifyMatchFailure(
+              binder.op, "Not converting to AtenSplitWithSizesOp due to input "
+                         "tensor mismatch");
+
+        if (binder.tensorListResultType(resultType) ||
+            binder.s64IntegerAttr(keepdims, "keepdims", 1) ||
+            binder.s64IntegerAttr(axis, "axis", 0))
+          return failure();
+
+        if (binder.op->getNumOperands() == 1)
+          return rewriter.notifyMatchFailure(
+              binder.op,
+              "No of operand should be 2. Keepdims is not yet implemented");
+
+        if (binder.tensorOperandAtIndex(split, 1))
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "split size is not specified");
+
+        Value vaxis = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getType<Torch::IntType>(),
+            rewriter.getI64IntegerAttr(axis));
+
+        auto splitTy = cast<Torch::ValueTensorType>(split.getType());
+        auto splitSizes = splitTy.getSizes();
+
+        if (splitSizes[0] == Torch::kUnknownSize)
+          return rewriter.notifyMatchFailure(
+              binder.op, "Dynamic shapes for Split is not yet supported");
+
+        if (!splitTy || !splitTy.hasSizes())
+          return failure();
+
+        ArrayRef<int64_t> splitShape = splitTy.getSizes();
+        int64_t splitRank = splitShape.size();
+
+        if (splitRank <= 1) {
+          Value splitInt = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), split);
+          rewriter.replaceOpWithNewOp<Torch::AtenSplitTensorOp>(
+              binder.op, resultType, self, splitInt, vaxis);
```

Comment:
For the rank 1 case, you should add the check for that single dimension to be 1 too.